### PR TITLE
ENG-11001 Editable UI component for chart description

### PR DESCRIPTION
**Pull request by qv-pat**
### Frontend
- [x] Added description to front end via EditableTitle
- [x] Added updateChartDescriptionOrSaveSlice function to handle auto save when editing description
- [x] Added action and reducer for UPDATE_CHART_DESCRIPTION

### Backend
- [x] Modified core/views to only overwrite name and description and value exists in query param

### CODE REVIEW

File: superset/assets/src/explore/reducers/exploreReducer.js [hunk 1]
```diff
@@ -99,6 +99,13 @@ export default function exploreReducer(state = {}, action) {
         slice: updatedSlice,
       };
     },
+    [actions.UPDATE_CHART_DESCRIPTION]() {
```
Comment (yuguang) [inline superset/assets/src/explore/reducers/exploreReducer.js:102]: What does this part of the code do? I am not familiar with this syntax.

File: superset/views/core.py [hunk 2]
```diff
@@ -1436,7 +1437,8 @@ def save_or_overwrite_slice(
         slc.viz_type = form_data['viz_type']
         slc.datasource_type = datasource_type
         slc.datasource_id = datasource_id
-        slc.slice_name = slice_name
+        slc.slice_name = slice_name or slc.slice_name
```
Comment (yuguang) [inline superset/views/core.py:1440]: I am quite sure session.merge(slc) takes care of merging with the existing entry. Remove the 'or' clause and see if this feature works as intended.

**Review by yuguang — CHANGES_REQUESTED** [review]
Let's test out a small change and do UAT before merging.

File: superset/views/core.py (hunk 2 above)
Comment (qv-pat) [reply superset/views/core.py:1440]: It does, however if the user updates the description but not the title the query param is empty and this overwrites the chart name with `None`.

File: superset/assets/src/explore/reducers/exploreReducer.js (hunk 1 above)
Comment (qv-pat) [reply superset/assets/src/explore/reducers/exploreReducer.js:102]: This a reducer, used for handling state management in the redux store

https://redux.js.org/basics/reducers#handling-more-actions

**Review by yuguang — APPROVED** [review]

**Pull request merged**